### PR TITLE
[Merged by Bors] - Druid Opa Authorizer

### LINKED ISSUE #145: OPA Authorizer
As an admin I'd like all requests to Druid resources optionally be authorized via OpenPolicyAgent (OPA).

To support this in Druid we need to
- [ ] Create a project that implements the Druid `Authorizer` interface
- [ ] Add integration tests
- [ ] We need to establish a _runbook_ or similar so we don't forget to test/update the authorizer project when we upgrade Druid
- [ ] (Optional for now) We should open a ticket upstream and propose donating the code so it gets integrated into Druid itself
- [ ] We need examples and documentation covering how to use this feature
- [ ] Package the Authorizer in the Docker image
- [ ] Make all necessary (unknown to me at the moment if any are needed) changes to the operator to support it

I'm happy to split this up into multiple tickets instead of one big one.
I can help with that or whoever picks it up is welcome to do so.

### PR DESCRIPTION
**Pull request by fhennig**
## Description

This PR relies on a number of other parts listed down below. In the operator, changes were made to make use of a new OPA authorizer extension for druid.

Parts:
- The Druid OPA Authorizer extension: https://github.com/stackabletech/druid-opa-authorizer - Needs to be compiled and added to the docker image
- The Druid Docker image https://github.com/stackabletech/docker-images/pull/58 needs and update too, to include the new extension
- The operator itself (this PR) needs to be able to configure the new extension
- The integration tests: https://github.com/stackabletech/integration-tests/pull/164 a new integration test was added to test the authorization extension

fixes #145 

## Review Checklist
- [x] Code contains useful comments
- [x] (Integration-)Test cases added (or not applicable)
- [x] Documentation added (or not applicable)
- [x] Changelog updated (or not applicable)
- [x] Cargo.toml only contains references to git tags (not specific commits or branches)
- [ ] Helm chart can be installed and deployed operator works (or not applicable)

Once the review is done, comment `bors r+` (or `bors merge`) to merge. [Further information](https://bors.tech/documentation/getting-started/#reviewing-pull-requests)

### CODE REVIEW

File: rust/crd/src/lib.rs [hunk 1]
```diff
@@ -261,6 +264,13 @@ pub struct S3Spec {
     pub endpoint: Option<String>,
 }
 
+#[derive(Clone, Debug, Default, Deserialize, JsonSchema, Serialize)]
+#[serde(rename_all = "camelCase")]
+pub struct OpaSpec {
+    pub opa_config_map_name: String,
+    pub opa_druid_api_path: String,
+}
```
Comment (maltesander) [inline rust/crd/src/lib.rs:272]: We just removed this in Kafka. I would only go for the config map name. The api path will be fixed in code or may be overriden via overrides if neccessary (but we can just document whats the default path should be)

File: rust/operator-binary/src/druid_controller.rs [hunk 2]
```diff
@@ -150,6 +157,31 @@ pub async fn reconcile_druid(
...
+
+        // Add the API path to the end
+        Some(
+            opa_host.trim_end_matches('/').to_owned()
+                + "/"
+                + opa_spec.opa_druid_api_path.trim_start_matches('/'),
+        )
+    } else {
+        None
+    };
```
Comment (maltesander) [inline rust/operator-binary/src/druid_controller.rs:183]: We usally mount the env var now from the configmaps and work with the envar from there on (https://github.com/stackabletech/kafka-operator/blob/26a074abf60f9243540ac34076a97c713a89cba3/rust/operator/src/kafka_controller.rs#L520).

**Review by maltesander — COMMENTED** [review]
This deviates a little on how we do it in trino / kafka.

**Review by maltesander — COMMENTED** [review]
Just minor things. Going to test now

File: docs/modules/ROOT/pages/usage.adoc [hunk 3]
```diff
@@ -127,8 +127,46 @@ This allows to ingest data from accessible buckets already. To configure a bucke
...
+
+Druid can connect to an Open Policy Agent (OPA) instance for authorization policy decisions. You need to run an OPA instance to connect to, for which we refer to the https://docs.stackable.tech/opa/index.html[OPA Operator docs]. How you can write RegoRules for Druid is explained <<_defining_regorules, below>>.
+
+Once you have defined your rules, you need to configure the OPA cluster name and endpoint to use for Druid authorization requests. Add a section to the `spec` for OPA:
+
+[source,yaml]
+----
+opa:
+  opaConfigMapName: simple-opa <1>
+  opaDruidApiPath: v1/data/druid/allow <2>
```
[the remark this answers is not in the document]
Comment (fhennig) [reply docs/modules/ROOT/pages/usage.adoc]: This is fixed

**Review by maltesander — APPROVED** [review]
LGTM.

**Pull request closed without merging**